### ISSUE TITLE
Retrieving 3rd party cookies

### ISSUE THREAD

**Issue by nreoch25** [amir20/phantomjs-node#651]
### Expected behavior

I need to retrieve 3rd party cookies.

In a previous version ( v0.7.2 ) I was able to access 3rd party cookies using:
`ph.getCookies(function(cookies) {
   console.log(cookies)
  });`

### Actual behavior
The getCookies function is now undefined in the phantom object ( v4.0.2 )
I tried to use:
`_page.property("cookies");`
This doesn't seem to return any third party cookies and usually just returns an empty array.
Is there something I'm missing with accessing cookies.
I'm wondering if accessing third party cookies is still possible?

Thanks in advance.

### Environment

- OS: Mac
- Node version (`node -v`):v6.9.2
- package version: 4.0.2

**Comment by amir20** [comment]
I have used `page.property('cookies');` and it works. See tests here https://github.com/amir20/phantomjs-node/blob/a0e550b1c61f87bc854697a75b7ee2f2a0c87186/src/__test__/page.spec.js#L169.

**Comment by nreoch25** [comment]
I'll try this again but when I used page.property("cookies") it didn't return 3rd party cookies. Maybe I was doing it wrong. Thanks for your help.

**Comment by nreoch25** [comment]
I tested it again and it only gets 1st party cookies through `page.property("cookies")`
Looks like I need to stick with version 1 which has a getCookie function on the phantom object that returns 1st and 3rd party cookies.
Unless you have any other ideas?  Thanks again.

**Comment by amir20** [comment]
Do you know what is the right call in vanilla phantomjs? I think it would call `page.cookies` and according to the [docs](http://phantomjs.org/api/phantom/property/cookies.html) it should work since version 1.7. There is no difference between v1 and v4 for this. So either you found a bug or there is something going on with your setup. 

Couple of things I wonder:
- Can you first try to see what phantomjs does? Does it even return third party cookies?
- Give me a full example that I can run locally? 
- Try running with `DEBUG=true node test.js` so that I can see the output.

**Comment by nreoch25** [comment]
Hey, thanks for re-opening.  So I can get third party cookies using the following:
phantom: v0.7.2, phantomjs-prebuilt: v2.1.4.
phantom: v0.7.2 has a getCookies method on the ph.
`phantom.create((ph) => {
    console.log("ph");
 }`
ph contain the getCookies method ( which returns 3rd party cookies ). This method is available up to version 1 but not 2.  The reason why I use v0.7.2 is because it doesn't come with phantomjs as dependency and I can install a newer version whereas v1 comes with phantomjs 1.9 and the setup was very buggy for third party cookies.

Here's the example I created using phantom v4.0.2. Returns only first party cookies.
`var phantom = require("phantom");
var _ph, _page;

phantom.create().then(ph => {
  _ph = ph;
  return _ph.createPage();
}).then(page => {
  _page = page;
  return _page.open("http://cbc.ca")
}).then(status => {
  return _page.property("cookies")
}).then(cookies => {
  console.log(cookies);
  //_page.close();
  //_ph.exit();
}).catch(e => console.log(e));
`
The output from running DEBUG=true node test on that is:

`debug: Sending: {"id":1,"target":"phantom","name":"createPage","params":[],"deferred":null}
debug: Sending NOOP command.
debug: Parsing: {"id":1,"target":"phantom","name":"createPage","params":[],"deferred":null,"response":{"pageId":1}}
debug: Sending: {"id":2,"target":"page$1","name":"invokeAsyncMethod","params":["open","http://cbc.ca"],"deferred":null}
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
info: TypeError: undefined is not a constructor (evaluating 'CBC.APP.SC.LinkTracker.trackArea')
info:   http://www.cbc.ca/:2175 in global code
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Parsing: {"id":2,"target":"page$1","name":"invokeAsyncMethod","params":["open","http://cbc.ca"],"deferred":null,"response":"success"}
debug: Sending: {"id":3,"target":"page$1","name":"property","params":["cookies"],"deferred":null}
debug: Received NOOP command.
debug: Sending NOOP command.
debug: Parsing: {"id":3,"target":"page$1","name":"property","params":["cookies"],"deferred":null,"response":[{"domain":".cbc.ca","expires":"Sat, 27 Apr 2019 14:24:31 GMT","expiry":1556375071,"httponly":false,"name":"__gads","path":"/","secure":false,"value":"ID=9b79b039062298fb:T=1493303071:S=ALNI_MYIDeJ3QTbe11w1xkMPdmiqPg72tw"},{"domain":".cbc.ca","httponly":false,"name":"s_sess","path":"/","secure":false,"value":"%20s_cc%3Dtrue%3B%20SC_LINKS%3D%3B%20s_sq%3D%3B"},{"domain":".cbc.ca","expires":"Fri, 27 Apr 2018 14:24:31 GMT","expiry":1524839071,"httponly":false,"name":"s_pers","path":"/","secure":false,"value":"%20s_nr2%3D1493303071425-New%7C1524839071425%3B"},{"domain":".cbc.ca","expires":"Fri, 27 Apr 2018 14:24:31 GMT","expiry":1524839071,"httponly":false,"name":"cX_P","path":"/","secure":false,"value":"j1wdg918blnl1qwo"},{"domain":".cbc.ca","httponly":false,"name":"cX_S","path":"/","secure":false,"value":"j20i3byzpr7j3d2w"},{"domain":"www.cbc.ca","expires":"Thu, 27 Apr 2017 14:54:30 GMT","expiry":1493304870,"httponly":false,"name":"_cb_svref","path":"/","secure":false,"value":"null"},{"domain":"www.cbc.ca","expires":"Sun, 27 May 2018 14:24:30 GMT","expiry":1527431070,"httponly":false,"name":"_chartbeat2","path":"/","secure":false,"value":".1493053410963.1493303070938.1011.DmSTnBB17eOwEAAeCD_X4YfBFgdHg"},{"domain":"www.cbc.ca","expires":"Sun, 27 May 2018 14:24:30 GMT","expiry":1527431070,"httponly":false,"name":"_cb","path":"/","secure":false,"value":"Cs2Okd50EhItCI6j"},{"domain":"www.cbc.ca","expires":"Fri, 27 Apr 2018 14:24:30 GMT","expiry":1524839070,"httponly":false,"name":"CBC_ID","path":"/","secure":false,"value":"0f388c48-9451-45ce-50eb-d5e96d97d897"},{"domain":"www.cbc.ca","expires":"Thu, 04 May 2017 14:24:30 GMT","expiry":1493907870,"httponly":false,"name":"_vff","path":"/","secure":false,"value":"www%2Ecbc%2Eca.00000000-0000-4000-8000-082f79d5b829.386ff7d437c6b39dd1ff2aff68b1a6fb.true"},{"domain":"www.cbc.ca","expires":"Sat, 27 Apr 2019 14:24:30 GMT","expiry":1556375070,"httponly":false,"name":"_vfa","path":"/","secure":false,"value":"www%2Ecbc%2Eca.00000000-0000-4000-8000-082f79d5b829.a23de7c4-5a0f-4e3b-b1d8-396f01ae3d8c.1493303070.1493303070.1493303070.1"},{"domain":"www.cbc.ca","expires":"Thu, 27 Apr 2017 14:30:30 GMT","expiry":1493303430,"httponly":false,"name":"_vfz","path":"/","secure":false,"value":"www%2Ecbc%2Eca.00000000-0000-4000-8000-082f79d5b829.1493303070.1.medium=direct|source=|terms="},{"domain":"www.cbc.ca","expires":"Thu, 27 Apr 2017 14:54:30 GMT","expiry":1493304870,"httponly":false,"name":"_vfb","path":"/","secure":false,"value":"www%2Ecbc%2Eca.00000000-0000-4000-8000-082f79d5b829.1.10.1493303070..00000000-0000-4000-8000-065dd08f2d53"},{"domain":".cbc.ca","expires":"Sat, 27 May 2017 14:24:29 GMT","expiry":1495895069,"httponly":false,"name":"cbclocal","path":"/","secure":false,"value":"weather%3Ds0000458%26radio%3Dtoronto%26news%3Dtoronto%26tv%3Dtoronto%26region%3Dtoronto"},{"domain":".cbc.ca","expires":"Tue, 09 Apr 2019 14:24:29 GMT","expiry":1554819869,"httponly":false,"name":"tvCenter","path":"/","secure":false,"value":"toronto"},{"domain":".cbc.ca","expires":"Tue, 09 Apr 2019 14:24:29 GMT","expiry":1554819869,"httponly":false,"name":"radioCenter","path":"/","secure":false,"value":"toronto"},{"domain":".cbc.ca","expires":"Tue, 09 Apr 2019 14:24:29 GMT","expiry":1554819869,"httponly":false,"name":"newsCenter","path":"/","secure":false,"value":"toronto"},{"domain":".cbc.ca","expires":"Tue, 09 Apr 2019 14:24:29 GMT","expiry":1554819869,"httponly":false,"name":"weatherCenter","path":"/","secure":false,"value":"s0000458"},{"domain":"www.cbc.ca","expires":"Sun, 27 May 2018 14:24:28 GMT","expiry":1527431068,"httponly":false,"name":"_cb_ls","path":"/","secure":false,"value":"1"},{"domain":"www.cbc.ca","httponly":false,"name":"akaas_feed","path":"/","secure":false,"value":"2147483647~rv=41~id=30155538af0d751bfc25897e70a659ef"}]}
[ { domain: '.cbc.ca',
    expires: 'Sat, 27 Apr 2019 14:24:31 GMT',
    expiry: 1556375071,
    httponly: false,
    name: '__gads',
    path: '/',
    secure: false,
    value: 'ID=9b79b039062298fb:T=1493303071:S=ALNI_MYIDeJ3QTbe11w1xkMPdmiqPg72tw' },
  { domain: '.cbc.ca',
    httponly: false,
    name: 's_sess',
    path: '/',
    secure: false,
    value: '%20s_cc%3Dtrue%3B%20SC_LINKS%3D%3B%20s_sq%3D%3B' },
  { domain: '.cbc.ca',
    expires: 'Fri, 27 Apr 2018 14:24:31 GMT',
    expiry: 1524839071,
    httponly: false,
    name: 's_pers',
    path: '/',
    secure: false,
    value: '%20s_nr2%3D1493303071425-New%7C1524839071425%3B' },
  { domain: '.cbc.ca',
    expires: 'Fri, 27 Apr 2018 14:24:31 GMT',
    expiry: 1524839071,
    httponly: false,
    name: 'cX_P',
    path: '/',
    secure: false,
    value: 'j1wdg918blnl1qwo' },
  { domain: '.cbc.ca',
    httponly: false,
    name: 'cX_S',
    path: '/',
    secure: false,
    value: 'j20i3byzpr7j3d2w' },
  { domain: 'www.cbc.ca',
    expires: 'Thu, 27 Apr 2017 14:54:30 GMT',
    expiry: 1493304870,
    httponly: false,
    name: '_cb_svref',
    path: '/',
    secure: false,
    value: 'null' },
  { domain: 'www.cbc.ca',
    expires: 'Sun, 27 May 2018 14:24:30 GMT',
    expiry: 1527431070,
    httponly: false,
    name: '_chartbeat2',
    path: '/',
    secure: false,
    value: '.1493053410963.1493303070938.1011.DmSTnBB17eOwEAAeCD_X4YfBFgdHg' },
  { domain: 'www.cbc.ca',
    expires: 'Sun, 27 May 2018 14:24:30 GMT',
    expiry: 1527431070,
    httponly: false,
    name: '_cb',
    path: '/',
    secure: false,
    value: 'Cs2Okd50EhItCI6j' },
  { domain: 'www.cbc.ca',
    expires: 'Fri, 27 Apr 2018 14:24:30 GMT',
    expiry: 1524839070,
    httponly: false,
    name: 'CBC_ID',
    path: '/',
    secure: false,
    value: '0f388c48-9451-45ce-50eb-d5e96d97d897' },
  { domain: 'www.cbc.ca',
    expires: 'Thu, 04 May 2017 14:24:30 GMT',
    expiry: 1493907870,
    httponly: false,
    name: '_vff',
    path: '/',
    secure: false,
    value: 'www%2Ecbc%2Eca.00000000-0000-4000-8000-082f79d5b829.386ff7d437c6b39dd1ff2aff68b1a6fb.true' },
  { domain: 'www.cbc.ca',
    expires: 'Sat, 27 Apr 2019 14:24:30 GMT',
    expiry: 1556375070,
    httponly: false,
    name: '_vfa',
    path: '/',
    secure: false,
    value: 'www%2Ecbc%2Eca.00000000-0000-4000-8000-082f79d5b829.a23de7c4-5a0f-4e3b-b1d8-396f01ae3d8c.1493303070.1493303070.1493303070.1' },
  { domain: 'www.cbc.ca',
    expires: 'Thu, 27 Apr 2017 14:30:30 GMT',
    expiry: 1493303430,
    httponly: false,
    name: '_vfz',
    path: '/',
    secure: false,
    value: 'www%2Ecbc%2Eca.00000000-0000-4000-8000-082f79d5b829.1493303070.1.medium=direct|source=|terms=' },
  { domain: 'www.cbc.ca',
    expires: 'Thu, 27 Apr 2017 14:54:30 GMT',
    expiry: 1493304870,
    httponly: false,
    name: '_vfb',
    path: '/',
    secure: false,
    value: 'www%2Ecbc%2Eca.00000000-0000-4000-8000-082f79d5b829.1.10.1493303070..00000000-0000-4000-8000-065dd08f2d53' },
  { domain: '.cbc.ca',
    expires: 'Sat, 27 May 2017 14:24:29 GMT',
    expiry: 1495895069,
    httponly: false,
    name: 'cbclocal',
    path: '/',
    secure: false,
    value: 'weather%3Ds0000458%26radio%3Dtoronto%26news%3Dtoronto%26tv%3Dtoronto%26region%3Dtoronto' },
  { domain: '.cbc.ca',
    expires: 'Tue, 09 Apr 2019 14:24:29 GMT',
    expiry: 1554819869,
    httponly: false,
    name: 'tvCenter',
    path: '/',
    secure: false,
    value: 'toronto' },
  { domain: '.cbc.ca',
    expires: 'Tue, 09 Apr 2019 14:24:29 GMT',
    expiry: 1554819869,
    httponly: false,
    name: 'radioCenter',
    path: '/',
    secure: false,
    value: 'toronto' },
  { domain: '.cbc.ca',
    expires: 'Tue, 09 Apr 2019 14:24:29 GMT',
    expiry: 1554819869,
    httponly: false,
    name: 'newsCenter',
    path: '/',
    secure: false,
    value: 'toronto' },
  { domain: '.cbc.ca',
    expires: 'Tue, 09 Apr 2019 14:24:29 GMT',
    expiry: 1554819869,
    httponly: false,
    name: 'weatherCenter',
    path: '/',
    secure: false,
    value: 's0000458' },
  { domain: 'www.cbc.ca',
    expires: 'Sun, 27 May 2018 14:24:28 GMT',
    expiry: 1527431068,
    httponly: false,
    name: '_cb_ls',
    path: '/',
    secure: false,
    value: '1' },
  { domain: 'www.cbc.ca',
    httponly: false,
    name: 'akaas_feed',
    path: '/',
    secure: false,
    value: '2147483647~rv=41~id=30155538af0d751bfc25897e70a659ef' } ]`

Thanks in advance for you help on this.

**Comment by amir20** [comment]
A couple of things I researched: 

1. A few people had complained that they wish to include their own phantomjs bin. I didn't know which one was worst, force a specific version or attempt to fix every bug with every version of phantomjs. I opted to force the version because that way I know my tests in travis work correctly. However, I am open to ideas of moving `phantomjs-prebuilt` to `peerDependency`. I haven't played with this. 
2. I looked at the code for v1. There looks to be two instances of `getCookies` [here](https://github.com/amir20/phantomjs-node/blob/v1/shim.coffee#L120). First of all, wtf. I never knew `phantom.cookies` was a thing.
3. Can you verify the above using phantomjs without node? 
4. If my theory is correct, then you should be able to call `phantom.cookies` via `phantomInstance.execute('phantom', 'property', ['cookies']);`. That's a little ugly if it works and I can probably just add `phantom.property('cookies')`. or just `phantom.cookies`. 

So to summarize, there seems to be `phantom.cookies` and `page.cookies` in v1. This is news to me. If you try option 4 and that works then we can try to add a PR to fix it. 

Please give above a try and let me know.

**Comment by nreoch25** [comment]
Hey Thanks for that. I will try that tonight for sure and get back to you.  I really appreciate your help on this.  Also, yes, when I have my own instance of phantom ( say via homebrew ) it works.

**Comment by nreoch25** [comment]
Hey, yes that works!! This is the code:
`var phantom = require("phantom");
var _ph, _page;
phantom.create().then(ph => {
  _ph = ph;
  return _ph.createPage();
}).then(page => {
  _page = page;
  return _page.open("http://cbc.ca")
}).then(status => {
  return _ph.execute("phantom", "property", ["cookies"])
}).then(cookies => {
  console.log(cookies);
  //_page.close();
  //_ph.exit();
}).catch(e => console.log(e));`

And the cookies get logged out and you can see the third party cookies:
`[ { domain: '.eyereturn.com',
    expires: 'Sun, 28 Apr 2019 01:16:54 GMT',
    expiry: 1556414214,
    httponly: false,
    name: 'erBeh',
    path: '/',
    secure: false,
    value: 'AwARdQAVAA==' },
  { domain: '.eyereturn.com',
    expires: 'Sun, 28 Apr 2019 01:16:54 GMT',
    expiry: 1556414214,
    httponly: false,
    name: 'erTok',
    path: '/',
    secure: false,
    value: 'AwAAAACyehYAjNZSAAEAAA==' },
  { domain: '.eyereturn.com',
    expires: 'Sun, 28 Apr 2019 01:16:54 GMT',
    expiry: 1556414214,
    httponly: false,
    name: 'er_guid',
    path: '/',
    secure: false,
    value: '7f3fc59e-ecef-4bf2-bfc5-9eeceffbf221' },
  { domain: '.doubleclick.net',
    expires: 'Sun, 28 Apr 2019 01:16:54 GMT',
    expiry: 1556414214,
    httponly: true,
    name: 'IDE',
    path: '/',
    secure: false,
    value: 'AHWqTUmzA0U0mfDtDbCYXOToL3bPbDdMCYAuM-9eUdqfeRIkH9Cq8BvuNA' },
  { domain: '.doubleclick.net',
    expires: 'Sun, 28 Apr 2019 01:16:54 GMT',
    expiry: 1556414214,
    httponly: false,
    name: 'id',
    path: '/',
    secure: false,
    value: '2281946c1c36002d||t=1493342214|et=730|cs=002213fd48bd243527f59f0198' },
  { domain: '.cbc.ca',
    expires: 'Sun, 28 Apr 2019 01:16:54 GMT',
    expiry: 1556414214,
    httponly: false,
    name: 's_vi',
    path: '/',
    secure: false,
    value: '[CS]v1|2C814C03051D24E2-6000013220008897[CE]' },
  { domain: '.cbc.ca',
    expires: 'Sun, 28 Apr 2019 01:16:54 GMT',
    expiry: 1556414214,
    httponly: false,
    name: '__gads',
    path: '/',
    secure: false,
    value: 'ID=5ac4986ef58e5f17:T=1493342214:S=ALNI_MaA6ZiH-qdbFntjdq0c7nVqta6KnA' },
  { domain: '.viafoura.co',
    expires: 'Sat, 29 Apr 2017 01:06:54 GMT',
    expiry: 1493428014,
    httponly: false,
    name: 'VfSess',
    path: '/',
    secure: false,
    value: 'rp698q2fciitvgtmt05osdvquq' },
  { domain: '.viafoura.co',
    expires: 'Mon, 26 Apr 2027 01:16:54 GMT',
    expiry: 1808702214,
    httponly: false,
    name: 'c',
    path: '/',
    secure: false,
    value: '1' },
  { domain: '.viafoura.co',
    expires: 'Sat, 28 Apr 2018 01:16:54 GMT',
    expiry: 1524878214,
    httponly: false,
    name: 'vfDeviceId',
    path: '/',
    secure: false,
    value: 'e04c66c5-4c57-4f74-820f-ee660ccc3d3b' },
  { domain: '.cxense.com',
    httponly: false,
    name: 'gcks',
    path: '/',
    secure: false,
    value: 'j215eac6lduyd4qo' },
  { domain: '.cxense.com',
    expires: 'Sat, 28 Apr 2018 01:16:54 GMT',
    expiry: 1524878214,
    httponly: true,
    name: 'gckp',
    path: '/',
    secure: false,
    value: '3rty67cxtxn2528i6p0wbmavl5' },
  { domain: '.cbc.ca',
    httponly: false,
    name: 's_sess',
    path: '/',
    secure: false,
    value: '%20s_cc%3Dtrue%3B%20SC_LINKS%3D%3B%20s_sq%3D%3B' },
  { domain: '.cbc.ca',
    expires: 'Sat, 28 Apr 2018 01:16:54 GMT',
    expiry: 1524878214,
    httponly: false,
    name: 's_pers',
    path: '/',
    secure: false,
    value: '%20s_nr2%3D1493342214429-New%7C1524878214429%3B' },
  { domain: 'www.cbc.ca',
    expires: 'Fri, 05 May 2017 01:16:54 GMT',
    expiry: 1493947014,
    httponly: false,
    name: '_vff',
    path: '/',
    secure: false,
    value: 'www%2Ecbc%2Eca.00000000-0000-4000-8000-082f79d5b829.e50b93473cff34239cc34510f45b2d23.true' },
  { domain: '.cbc.ca',
    expires: 'Sat, 28 Apr 2018 01:16:53 GMT',
    expiry: 1524878213,
    httponly: false,
    name: 'cX_P',
    path: '/',
    secure: false,
    value: 'j214rpnfzbz6dcjh' },
  { domain: '.cbc.ca',
    httponly: false,
    name: 'cX_S',
    path: '/',
    secure: false,
    value: 'j215eac6lduyd4qo' },
  { domain: '.casalemedia.com',
    expires: 'Sat, 29 Apr 2017 01:16:53 GMT',
    expiry: 1493428613,
    httponly: false,
    name: 'CMDD',
    path: '/',
    secure: false,
    value: 'AAJ0lwE*' },
  { domain: '.casalemedia.com',
    httponly: false,
    name: 'CMSC',
    path: '/',
    secure: false,
    value: 'WQKYBQ**' },
  { domain: '.casalemedia.com',
    expires: 'Sat, 29 Apr 2017 01:16:53 GMT',
    expiry: 1493428613,
    httponly: false,
    name: 'CMST',
    path: '/',
    secure: false,
    value: 'WQKYBVkCmAUB' },
  { domain: '.casalemedia.com',
    expires: 'Thu, 27 Jul 2017 01:16:53 GMT',
    expiry: 1501118213,
    httponly: false,
    name: 'CMPS',
    path: '/',
    secure: false,
    value: '169' },
  { domain: '.casalemedia.com',
    expires: 'Sat, 28 Apr 2018 01:16:53 GMT',
    expiry: 1524878213,
    httponly: false,
    name: 'CMID',
    path: '/',
    secure: false,
    value: 'WQKYBcAoJ30AAH7SEtUAAACb' },
  { domain: '.casalemedia.com',
    expires: 'Fri, 28 Apr 2017 02:16:53 GMT',
    expiry: 1493345813,
    httponly: false,
    name: 'CMTEST',
    path: '/',
    secure: false,
    value: '1' },
  { domain: 'www.cbc.ca',
    expires: 'Sun, 28 Apr 2019 01:16:53 GMT',
    expiry: 1556414213,
    httponly: false,
    name: '_vfa',
    path: '/',
    secure: false,
    value: 'www%2Ecbc%2Eca.00000000-0000-4000-8000-082f79d5b829.b663079b-d514-4d30-876d-c4059c35e7ec.1493342213.1493342213.1493342213.1' },
  { domain: 'www.cbc.ca',
    expires: 'Fri, 28 Apr 2017 01:22:53 GMT',
    expiry: 1493342573,
    httponly: false,
    name: '_vfz',
    path: '/',
    secure: false,
    value: 'www%2Ecbc%2Eca.00000000-0000-4000-8000-082f79d5b829.1493342213.1.medium=direct|source=|terms=' },
  { domain: 'www.cbc.ca',
    expires: 'Fri, 28 Apr 2017 01:46:53 GMT',
    expiry: 1493344013,
    httponly: false,
    name: '_vfb',
    path: '/',
    secure: false,
    value: 'www%2Ecbc%2Eca.00000000-0000-4000-8000-082f79d5b829.1.10.1493342213..00000000-0000-4000-8000-065dd08f2d53' },
  { domain: '.scorecardresearch.com',
    expires: 'Thu, 18 Apr 2019 01:16:53 GMT',
    expiry: 1555550213,
    httponly: false,
    name: 'UIDR',
    path: '/',
    secure: false,
    value: '1493342213' },
  { domain: '.scorecardresearch.com',
    expires: 'Thu, 18 Apr 2019 01:16:53 GMT',
    expiry: 1555550213,
    httponly: false,
    name: 'UID',
    path: '/',
    secure: false,
    value: '11324a15614023a451492cg1493342213' },
  { domain: 'www.cbc.ca',
    expires: 'Fri, 28 Apr 2017 01:46:52 GMT',
    expiry: 1493344012,
    httponly: false,
    name: '_cb_svref',
    path: '/',
    secure: false,
    value: 'null' },
  { domain: 'www.cbc.ca',
    expires: 'Mon, 28 May 2018 01:16:52 GMT',
    expiry: 1527470212,
    httponly: false,
    name: '_chartbeat2',
    path: '/',
    secure: false,
    value: '.1439523032652.1493342212909.0000000000000001.DroXDzBIKBEEDvB19NCcgk9lDrZY1f' },
  { domain: 'www.cbc.ca',
    expires: 'Mon, 28 May 2018 01:16:52 GMT',
    expiry: 1527470212,
    httponly: false,
    name: '_cb',
    path: '/',
    secure: false,
    value: 'Do5dQI9_v6FLuaRf' },
  { domain: 'www.cbc.ca',
    expires: 'Sat, 28 Apr 2018 01:16:52 GMT',
    expiry: 1524878212,
    httponly: false,
    name: 'CBC_ID',
    path: '/',
    secure: false,
    value: '94821e75-3c15-15dd-facb-4fe25f7eae9a' },
  { domain: '.cbc.ca',
    expires: 'Sun, 28 May 2017 01:16:52 GMT',
    expiry: 1495934212,
    httponly: false,
    name: 'cbclocal',
    path: '/',
    secure: false,
    value: 'weather%3Ds0000458%26radio%3Dtoronto%26news%3Dtoronto%26tv%3Dtoronto%26region%3Dtoronto' },
  { domain: '.cbc.ca',
    expires: 'Wed, 10 Apr 2019 01:16:52 GMT',
    expiry: 1554859012,
    httponly: false,
    name: 'tvCenter',
    path: '/',
    secure: false,
    value: 'toronto' },
  { domain: '.cbc.ca',
    expires: 'Wed, 10 Apr 2019 01:16:52 GMT',
    expiry: 1554859012,
    httponly: false,
    name: 'radioCenter',
    path: '/',
    secure: false,
    value: 'toronto' },
  { domain: '.cbc.ca',
    expires: 'Wed, 10 Apr 2019 01:16:52 GMT',
    expiry: 1554859012,
    httponly: false,
    name: 'newsCenter',
    path: '/',
    secure: false,
    value: 'toronto' },
  { domain: '.cbc.ca',
    expires: 'Wed, 10 Apr 2019 01:16:52 GMT',
    expiry: 1554859012,
    httponly: false,
    name: 'weatherCenter',
    path: '/',
    secure: false,
    value: 's0000458' },
  { domain: 'i.cbc.ca',
    httponly: false,
    name: 'akaas_feed',
    path: '/',
    secure: false,
    value: '2147483647~rv=62~id=b9a6127beb18cc98ce079bb99d93c492' },
  { domain: 'www.cbc.ca',
    expires: 'Mon, 28 May 2018 01:16:52 GMT',
    expiry: 1527470212,
    httponly: false,
    name: '_cb_ls',
    path: '/',
    secure: false,
    value: '1' },
  { domain: 'www.cbc.ca',
    httponly: false,
    name: 'akaas_feed',
    path: '/',
    secure: false,
    value: '2147483647~rv=71~id=1e244f3277faa8d71e9d3a8eef2d9474' } ]`

thanks for this! It's nice that I won't have to use an old version.

**Comment by amir20** [comment]
Nice. I'll keep this open and probably add an official method to make this work.

**Comment by amir20** [comment]
@nreoch25 I just pushed version `4.0.3` where it has `phantom.cookies()` and `page.cookies()`. Look at the tests.

**Comment by nreoch25** [comment]
Thanks @amir20 . This is a big help. I appreciate this.
